misc/version_updater.py

```diff
@@ -0,0 +1,325 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+"""
+Helper that help changing the version of a tool across the repository.
+"""
+
+import enum
+import re
+import sys
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from fileinput import FileInput
+from pathlib import Path
+from typing import Callable, Dict, List, Tuple, Union
+
+ReplacementPattern = Union[str, Callable[[str], str]]
+
+RawRegexes = List["Replace"]
+
+ROOT_DIR = (Path(__file__) / "../..").resolve()
+
+
+@dataclass
+class Replace:
+    regex: str
+    replaced: ReplacementPattern
```

Comment:
Most (maybe all?) of the expressions below share the text before the version number. 
I.e. the `"poetry-version"` here:

```python
Tool.Poetry: [Replace(r"poetry-version: [0-9.]+", "poetry-version: {version}")],
```

Is it not possible to use regex groups to avoid this duplication? Then the definitions will look more like

```python
Tool.Poetry: [Replace(r"poetry-version: ([0-9.]+)")],
```